Sources/SQLite/Core/Connection.swift

```diff
@@ -101,11 +101,21 @@ public final class Connection {
     ///
     ///     Default: `false`.
     ///
+    ///   - target: An optional target queue for database operations to be executed on.
+    ///     Set this to a custom serial queue if you need to schedule work that must be
+    ///     synchronized with respect to database operations.
+    ///
+    ///     **Important:** The target queue **must not** be a concurrent queue, or access
+    ///     to the returned connection is no longer guaranteed be thread-safe.
+    ///
+    /// - Throws: `Result.Error` iff a connection cannot be established.
+    ///
     /// - Returns: A new database connection.
-    public init(_ location: Location = .inMemory, readonly: Bool = false) throws {
+    public init(_ location: Location = .inMemory, readonly: Bool = false, target: DispatchQueue? = nil) throws {
+        queue = DispatchQueue(label: "SQLite.Database", target: target)
```

Comment:
looks like a queue is created here even though it might not be needed (when `target` is set)?

Reply:
Targeted queue hierarchies are meant to be very cheap, so this is intentional.